src/main/java/org/testng/junit/JUnitTestRunner.java

```diff
@@ -99,7 +96,13 @@ public void endTest(Test test) {
 
     org.testng.internal.TestResult tr = recordResults(test, tri);
 
-    TestListenerHelper.runTestListeners(tr, m_parentRunner.getTestListeners());
+    if (tr.getStatus() == ITestResult.STARTED) {
+      TestListenerHelper.runTestListeners(tr, m_parentRunner.getTestListeners()); 
+      } else {
+        List<ITestListener> test_Listeners_reverted = Lists.newArrayList(m_parentRunner.getTestListeners());

```

Comment:
Please have `test_Listeners_reverted` renamed to `testListenersReversed`

Reply:
Done